Failed to compile GRPC under SLES11: pthread_setname_np is missing

GRPC 1.39.1 failed to compile under SLES11 because pthread_setname_np was introduced in glibc 2.12.

Glibc version used in SLES11 is libc 2.11.3
    bash-3.2$ getconf GNU_LIBC_VERSION
    glibc 2.11.3


### What version of gRPC and what language are you using?
1.39.1 / C++

### What operating system (Linux, Windows,...) and version?
Linux, SLES11

### What runtime / compiler are you using (e.g. python version or version of gcc)
GCC 10

### What did you do?
Please provide either 1) A unit test for reproducing the bug or 2) Specific steps for us to follow to reproduce the bug. If there’s not enough information to debug the problem, gRPC team may close the issue at their discretion. You’re welcome to re-open the issue once you have a reproduction.
Compiling GRPC with old GLIBC give an error

### What did you expect to see?
GRPC compiled on old GLIBC.

### What did you see instead?
/users/xxx/.conan/data/grpc/1.39.1/_/_/build/23a963e55e4a6da15f34448b8aceb08789fb8b92/source_subfolder/src/core/lib/gprpp/thd_posix.cc:125:25: error: ‘pthread_setname_np’ was not declared in this scope; did you mean ‘pthread_getattr_np’?
  125 |                         pthread_setname_np(pthread_self(), buf);
      |                         ^~~~~~~~~~~~~~~~~~
      |                         pthread_getattr_np


Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

### Anything else we should know about your project / environment?
Compile GRPC with Conan on SLES11.

--- a/include/grpc/impl/codegen/port_platform.h
+++ b/include/grpc/impl/codegen/port_platform.h
@@ -187,7 +187,7 @@
 #ifdef __GLIBC__
 #define GPR_POSIX_CRASH_HANDLER 1
 #ifdef __GLIBC_PREREQ
-#if __GLIBC_PREREQ(2, 11)
+#if __GLIBC_PREREQ(2, 12)
 #define GPR_LINUX_PTHREAD_NAME 1
 #endif
 #else

--- a/include/grpc/impl/codegen/port_platform.h
+++ b/include/grpc/impl/codegen/port_platform.h
@@ -186,7 +186,14 @@
 #endif /* _LP64 */
 #ifdef __GLIBC__
 #define GPR_POSIX_CRASH_HANDLER 1
+#ifdef __GLIBC_PREREQ
+#if __GLIBC_PREREQ(2, 11)
 #define GPR_LINUX_PTHREAD_NAME 1
+#endif
+#else
+// musl libc & others
+#define GPR_LINUX_PTHREAD_NAME 1
+#endif
 #include <linux/version.h>
 #else /* musl libc */
 #define GPR_MUSL_LIBC_COMPAT 1

